### PR TITLE
Http Refactoring

### PR DESCRIPTION
**Pull request by dolanbernard**
### Motivation

We would like to provide the option for an android-sdk build without a dependency on GRPC. This PR includes numerous refactoring changes which remove dependencies on GRPC classes within code that does not utilize GRPC.

### In this PR
* Removing dependencies on io.grpc.Status in HTTP code
* Removing dependencies on io.grpc.StatusRuntimeException in HTTP code
* All network exceptions are now checked exceptions and include failure reason
* All NetworkExceptions caused by other Exceptions will include cause

### Future Work
* Provide optional build configuration which excludes all GRPC classes

### CODE REVIEW

File: android-sdk/src/androidTest/java/com/mobilecoin/lib/BlockchainClientTest.java [hunk 1]
```diff
@@ -79,7 +82,8 @@ public void clientRespectsCacheTTL() throws Exception {
         BlockchainClient blockchainClient = new BlockchainClient(
                 RandomLoadBalancer.create(consensusUri),
                 Environment.getTestFogConfig().getClientConfig().consensus,
-                Duration.ofMillis(1));
+                Duration.ofMillis(1),
+                TransportProtocol.forGRPC());
```
[the remark this answers is not in the document]
Comment (dolanbernard) [reply android-sdk/src/androidTest/java/com/mobilecoin/lib/BlockchainClientTest.java]: Yup, this is fixed here and in every client instantiation in tests now.

File: android-sdk/src/androidTest/java/com/mobilecoin/lib/TestFogConfig.java [hunk 2]
```diff
@@ -123,15 +149,18 @@ private static TestFogConfig getFogConfig(Environment.TestEnvironment testEnviro
             case MOBILE_DEV:
                 return new TestFogConfig(fogUri, consensusUris, TEST_USERNAME,
                         TEST_PASSWORD, getDevClientConfig(storageAdapter),
-                        mobiledevFogAuthoritySpki, "");
+                        mobiledevFogAuthoritySpki, "", TransportProtocol.forGRPC(),
+                        new GRPCServiceAPIManager(), new RestServiceAPIManager());
```
[the remark this answers is not in the document]
Comment (dolanbernard) [reply android-sdk/src/androidTest/java/com/mobilecoin/lib/TestFogConfig.java]: This is leftover from a reverted MobileCoinClient change. It's removed now.

File: android-sdk/src/androidTest/java/com/mobilecoin/lib/MobileCoinClientTest.java [hunk 3]
```diff
@@ -204,20 +198,14 @@ public void test_bad_trust_root_must_fail() throws InvalidFogResponse,
...
+        MobileCoinClient mobileCoinClient = MobileCoinClientBuilder.newBuilder()
+                .setAccountKey(TestKeysManager.getNextAccountKey())
+                .setTestFogConfig(fogConfig)
+                .build();
         try {
             mobileCoinClient.getBalance();
             Assert.fail("Invalid trust root must fail the test");
-        } catch (NetworkException ex) {
+        } catch (NetworkException | AttestationException ex) {
             // success
```
[the remark this answers is not in the document]
Comment (dolanbernard) [reply android-sdk/src/androidTest/java/com/mobilecoin/lib/MobileCoinClientTest.java]: Good catch. I removed the AttestationException. This was most likely something temporary I added to fix a build while refactoring Exception flow.

**Review by voloshyn — APPROVED** [review]
LGTM!

**Pull request merged**